diff --git a/cvs2svn_rcsparse/default.py b/cvs2svn_rcsparse/default.py
index 24825e9..293a8fe 100644
--- a/cvs2svn_rcsparse/default.py
+++ b/cvs2svn_rcsparse/default.py
@@ -37,7 +37,7 @@ class _TokenStream:
     self.idx = 0
     self.buf = self.rcsfile.read(self.CHUNK_SIZE)
     if self.buf == '':
-      raise RuntimeError, 'EOF'
+      raise Exception("Sorry, EOF")
 
   def get(self):
     "Get the next token from the RCS file."
diff --git a/cvs2svn_rcsparse/run-tests.py b/cvs2svn_rcsparse/run-tests.py
index caa0fdc..65614d6 100755
--- a/cvs2svn_rcsparse/run-tests.py
+++ b/cvs2svn_rcsparse/run-tests.py
@@ -46,7 +46,7 @@ for filename in filelist:
     f = StringIO()
     try:
         parse(open(filename, 'rb'), LoggingSink(f))
-    except Exception, e:
+    except Exception as e:
         sys.stderr.write('Error parsing file: %s!\n' % (e,))
         all_tests_ok = 0
     else:
